address.js

```diff
@@ -191,6 +194,11 @@ define(function(require) {
         api.method('remove').body(body)()
       }
 
+      api.timeout = function(v) {
+        timeout = util.isValidNumber(v) ? v : timeout
```

Comment:
No no, the validation needs to happen in the middleware, where it's *used*. The point of validating where the value is being used is because even if it's validated here it can still be changed in transit. The validation needs to happen in this function, before the value is used: https://github.com/pandey-mohit/address/blob/9c92f228609002af6e9fd32df940463ff47311ea/middleware.js#L11

Reply:
IMHO, the validation needs to be happen at the first place. Also, I am not sure how it can be changed in transit.
But, I am happy to make changes and add it to middleware, since asked.

Reply:
You could add it to both, but it’d be overkill. Since the timeout is exposed on the request object it can be changed at any point during the request handling process, which of course includes the end request handler but it also includes all middleware, and whatever meddling that address does itself. It’d be trivial to write a middleware function that messes up the value, which is why validation at use is the only place it truly matters.

Throwing an error at that point however may be less convenient, since it very likely wouldn’t be in the same stack as where the value was originally set on the request object, which incidentally is an argument for doing the validation twice.